client/js/libs/handlebars/ircmessageparser/merge.js

```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+const anyIntersection = require("./anyIntersection");
+const fill = require("./fill");
+
+function Object_assign(target) {
+	Array.prototype.slice.call(arguments, 1).forEach(function(obj) {
+		Object.keys(obj).forEach(function(key) {
+			target[key] = obj[key];
+		});
+	});
+	return target;
+}
+
+function merge(textParts, styleFragments) {
+	const cleanText = styleFragments.map(fragment => fragment.text).join("");
+
+	const allParts = textParts.concat(fill(textParts, cleanText))
+        .sort((a, b) => a.start - b.start);
+
+	return allParts.map(textPart => {
+		textPart.fragments = styleFragments
+            .filter(fragment => anyIntersection(textPart, fragment))
```

Comment:
@Bonuspunkt, thanks a lot for your previous comments. That helped me commenting the code and understanding/reviewing the flow/edge cases of `findChannels`.
Would you be able to help me on the rest of this PR? Ideally, add a comment to lines that may seem confusing to someone who hasn't written this code to explain why this was done or necessary.

Thanks!!